### PR TITLE
Simplify running simulations 

### PR DESCRIPTION
**Pull request by boriskaus**
The current way to perform a simulation is a bit complicated, and not easily customisable. 

With this PR, we simplify that. The basic idea is that we have a single 2D code, `MTK_GeoParams_2D` that runs the main simulation:

```Julia
Grid, Arrays, Tracers, Dikes, time_props = MTK_GeoParams_2D(MatParam::Tuple, Num::NumParam, Dike_params::DikeParam); 
```
The input are the material parameters (from `GeoParams`) and two structures in which we specify the numerical parameters and the dike parameters. 

Within `MTK_GeoParams_2D` we have a number of callback functions:
```Julia
        # Add new dike every X years -----------------
        Tracers = MTK_inject_dikes(Grid, Num, Arrays, Mat_tup, Dikes, Tracers, dike, Tnew_cpu)
        # --------------------------------------------

        # Update time-dependent properties -----------
        MTK_update_TimeDepProps!(time_props, Grid, Num, Arrays, Mat_tup, Dikes)
        # --------------------------------------------

        # Visualize results --------------------------
        MTK_visualize_output(Grid, Num, Arrays, Mat_tup, Dikes)
        # --------------------------------------------
          
        # Save output to disk once in a while --------
        MTK_save_output(Grid, Arrays, Tracers, Dikes, time_props)
        # --------------------------------------------

        # Optionally Update arrays (such as T) -------
        MTK_update_Arrays!(Arrays, Grid, Num)
        # --------------------------------------------
        
        # Display output -----------------------------
        MTK_print_output(Grid, Num, Arrays, Mat_tup, Dikes)
        # --------------------------------------------
```
They are now set to default parameters (many are empty), but can be overwritten in a user-code as:

```Julia
import MagmaThermoKinematics.MTK_GMG_2D.MTK_print_output
function MTK_print_output(Grid::GridData, Num::NumericalParameters, Arrays::NamedTuple, Mat_tup::Tuple, Dikes::DikeParameters)
    println("$(Num.it), max. temperature = $(maximum(Arrays.Tnew))")
    return nothing
end
```

An example of how this works is given in `examples/MTK_GMG_2D_example2.jl` and models a case underneath Unzen:
Screenshot 2023-12-08 at 18 00 35

### CODE REVIEW

File: test/test_MTK_GMG_2D.jl [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+Topo_cart = load_GMG("../examples/Topo_cart")       # Note: Laacher seee is around [10,20]
+
+# Create 3D grid of the region
+X,Y,Z       =   XYZGrid(-23:.1:23,-19:.1:19,-20:.1:5)
+Data_set3D  =   CartData(X,Y,Z,(Phases=zeros(Int64,size(X)),Temp=zeros(size(X))));       # 3D dataset
+
+# Create 2D cross-section
+Nx          =   135;  # resolution in x
+Nz          =   135;
```
Comment (aelligp) [inline test/test_MTK_GMG_2D.jl:83]:
```suggestion
Nx          =   Num.Nx;  # resolution in x
Nz          =   Num.Nz;
```

**Review by albert-de-montserrat — CHANGES_REQUESTED** [review]
Looks overall good to me. As a general comment, the examples are somewhat hard to read, some suggestions:

1. Remove all the commented lines
2. Don't use short hand branches. For example, dont write 
```julia
    if USE_GPU; Phases      =   CUDA.ones(Int64,Grid.N...)
    else        Phases      =   ones(Int64,Grid.N...)   end
```
but 
```julia
    if USE_GPU
        Phases  = CUDA.ones(Int64,Grid.N...)
    else 
        Phases   =  ones(Int64,Grid.N...)   
    end
```
3. Don't stack two lines of code into one. E.g. split this into two lines of code
```julia
if isdir("viz3D_out")==false mkdir("viz3D_out") end; loadpath = "./viz3D_out/"; pvd = paraview_collection("Example3D");
```
4. Try to tabulate the equal assignments of the same blocks. E.g. as in here
```julia
            x_1d =  Grid.coord1D[1]/1e3;
            z_1d =  Grid.coord1D[2]/1e3;
            temp_data = Array(Arrays.Tnew)'
            ϕ_data = Array(Arrays.ϕ)'
            phase_data = Float64.(Array(Arrays.Phases))'
```

**Review by aelligp — CHANGES_REQUESTED** [review]
Some minor things e.g. typos

File: src/Diffusion.jl [hunk 2]
```diff
@@ -290,10 +290,10 @@ end
 
 In-place function that creates 3D arrays with `x`,`y` and `z` coordinates using the `Grid` information
 """
-function GridArray!(X::AbstractArray, Z::AbstractArray, Grid::GridData)
+function GridArray!(X::AbstractArray, Y::AbstractArray, Z::AbstractArray, Grid::GridData)
     @parallel (1:Grid.N[1], 1:Grid.N[2], 1:Grid.N[3]) GridArray!(X, Y, Z, Grid.coord1D[1], Grid.coord1D[2], Grid.coord1D[3])
 
-    return nothing
+    return
```
[the remark this answers is not in the document]
Comment (albert-de-montserrat) [reply src/Diffusion.jl:296]: Actually `return` and `return nothing` are equivalent :)

**Review by aelligp — APPROVED** [review]

**Pull request merged**